mlir/test/Quantum/ConversionTest.mlir

```diff
@@ -367,13 +367,47 @@ func.func @hamiltonian(%obs : !quantum.obs, %p1 : memref<1xf64>, %p2 : memref<3x
 // Measurements //
 //////////////////
 
-// CHECK: llvm.func @__catalyst__qis__Measure(!llvm.ptr<struct<"Qubit", opaque>>) -> !llvm.ptr<struct<"Result", opaque>>
+// CHECK: llvm.func @__catalyst__qis__Measure(!llvm.ptr<struct<"Qubit", opaque>>, i8) -> !llvm.ptr<struct<"Result", opaque>>
 
 // CHECK-LABEL: @measure
 func.func @measure(%q : !quantum.bit) -> !quantum.bit {
 
-    // CHECK: llvm.call @__catalyst__qis__Measure(%arg0)
-    %res, %new_q = quantum.measure %q : i1, !quantum.bit
+    // CHECK: [[postselect:%.+]] = llvm.mlir.constant({{.+}} : i8) : i8
+
+    // CHECK: llvm.call @__catalyst__qis__Measure(%arg0, [[postselect]])
+    %res, %new_q = quantum.measure %q {postselect = -1 : i8} : i1, !quantum.bit
+
+    // CHECK: return %arg0
+    return %new_q : !quantum.bit
+}
+
+// -----
+
+// CHECK: llvm.func @__catalyst__qis__Measure(!llvm.ptr<struct<"Qubit", opaque>>, i8) -> !llvm.ptr<struct<"Result", opaque>>
+
+// CHECK-LABEL: @measure
+func.func @measure(%q : !quantum.bit) -> !quantum.bit {
+
+    // CHECK: [[postselect:%.+]] = llvm.mlir.constant({{.+}} : i8) : i8
+
+    // CHECK: llvm.call @__catalyst__qis__Measure(%arg0, [[postselect]])
+    %res, %new_q = quantum.measure %q {postselect = 0 : i8} : i1, !quantum.bit
+
+    // CHECK: return %arg0
+    return %new_q : !quantum.bit
+}
+
+// -----
+
+// CHECK: llvm.func @__catalyst__qis__Measure(!llvm.ptr<struct<"Qubit", opaque>>, i8) -> !llvm.ptr<struct<"Result", opaque>>
+
+// CHECK-LABEL: @measure
+func.func @measure(%q : !quantum.bit) -> !quantum.bit {
+
+    // CHECK: [[postselect:%.+]] = llvm.mlir.constant({{.+}} : i8) : i8
+
+    // CHECK: llvm.call @__catalyst__qis__Measure(%arg0, [[postselect]])
+    %res, %new_q = quantum.measure %q {postselect = 1 : i8} : i1, !quantum.bit
```

Comment:
Now that postselect is required, I am not sure it is useful to have these additional tests. Happy to keep them, but they might be redundant.

Maybe another test is what happens if the MLIR contains a `postselect = -2`? We should emit an error.

Reply:
Is this error check necessary here? I mean, at the MLIR, `postselect` can contain any `i8` value. But as far as I understand, this test generates the LLVM code lowering, but that code is not lowered further down, right? The error is raised when `__catalyst__qis__Measure` is called, but this is not going to happen in this test. Please, correct me if I am wrong.

Reply:
MLIR allows you to easily specify constraints on attributes, should you be able to declare a type constraint of "positive i8 less than 2" right in the tablegen declaration.

Reply:
Thanks for the hint. Now the dialect looks like this:

`OptionalAttr<ConfinedAttr<ConfinedAttr<I8Attr, [IntMinValue<-1>]>, [IntMaxValue<1>]>>:$postselect`